### PR TITLE
Fix required opt validation bugs in CLI

### PR DESCRIPTION
**Pull request by ghubstan**
Many required CLI opt checks were incorrectly passing due to assignment of `""` default values.  Server side validation was helping cover up some of affects of this type of bug, but this change resolves the issue in the CLI side.

Credit to @BtcContributor for exposing the problem while beta testing the api.

Changes are listed below.

Unsets empty string default values on some the `createoffer` method's required option declarations in CreateOfferOptionParser, and adds a check for presence of required currency-code option.

- Removes default "" value from required opt direction.
- Removes default "" value from required opt currency-code.
- Removes default "" value from required opt amount.
- Removes default "" value from required opt min-amount.
- Removes default "" value from required opt fixed-price.
- Removes default "" value from required opt security-deposit.
- Check for required currency-code option.

Unsets empty string default values on some of the `takeoffer` method's required option declarations in TakeOfferOptionParser.

- Removes default "" value from required opt offer-id.
- Removes default "" value from required opt payment-account.

Other opt parser default values removed from:

- CancelOfferOptionParser#offer-id
- CreatePaymentAcctOptionParser#payment-account-form
- GetAddressBalanceOptionParser#address
- GetBTCMarketPriceOptionParser#currency-code
- GetOfferOptionParser#offer-id
- GetOffersOptionParser#direction
- GetOffersOptionParser#currency-code
- GetPaymentAcctFormOptionParser#payment-method-id
- GetTradeOptionParser#trade-id
- GetTransactionOptionParser#transaction-id
- RegisterDisputeAgentOptionParser#registration-key
- RegisterDisputeAgentOptionParser#dispute-agent-type
- RemoveWalletPasswordOptionParser#wallet-password
- SendBsqOptionParser#address
- SendBsqOptionParser#amount
- SendBtcOptionParser#address
- SendBtcOptionParser#amount
- SetTxFeeRateOptionParser#tx-fee-rate
- SetWalletPasswordOptionParser#wallet-password
- UnlockWalletOptionParser#wallet-password
- UnlockWalletOptionParser#timeout
- WithdrawFundsOptionParser#trade-id
- WithdrawFundsOptionParser#address

PR https://github.com/bisq-network/bisq/pull/5273 should be reviewed & merged before this one.

### CODE REVIEW

**Review by ghubstan — COMMENTED** [review]
The last part of commit d01a7b7feec7167b66ec808529d2e63ffe248442 's message is incorrect.  Commit d01a7b7feec7167b66ec808529d2e63ffe248442 makes _empty_ opt values invalid, and throws an exception with the conventional `no <opt> specified` error message. Before this change, the CLI's opt parser accepted `--offer-id=` as a valid option, with an empty value.

The _next_ commit makes the option parser throw option parser exceptions due to  _missing (not empty)_ option values.  The next commit will result in a stylistically consistent error message printed to the console when `--offer-id` is present without an `=` delimiter (as if it were a no-arg option).

File: cli/src/test/java/bisq/cli/opt/OptionParsersTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        String[] args = new String[]{
+                PASSWORD_OPT,
+                createoffer.name(),
+                "--" + OPT_PAYMENT_ACCOUNT + "=" + "abc-payment-acct-id-123",
+                "--" + OPT_DIRECTION + "=" + "BUY",
+                "--" + OPT_CURRENCY_CODE + "=" + "EUR",
+                "--" + OPT_AMOUNT + "=" + "0.125",
+                "--" + OPT_MKT_PRICE_MARGIN + "=" + "0.0",
+                "--" + OPT_SECURITY_DEPOSIT + "=" + "25.0"
+        };
```
Comment (sqrrm) [inline cli/src/test/java/bisq/cli/opt/OptionParsersTest.java:132]: Is there an assert missing here?

Comment (ghubstan) [reply cli/src/test/java/bisq/cli/opt/OptionParsersTest.java:132]: Added in commit 70da6d19fa694c5ae11212645fdfa4ea47529854.

Comment (ghubstan) [reply cli/src/test/java/bisq/cli/opt/OptionParsersTest.java:132]: In next PR ([5284](https://github.com/bisq-network/bisq/pull/5284)) more asserts were added to check valid, parsed opt values in commit 389abf4.

```
	assertEquals("abc-payment-acct-id-123", parser.getPaymentAccountId());
	assertEquals("BUY", parser.getDirection());
	assertEquals("EUR", parser.getCurrencyCode());
	assertEquals("0.125", parser.getAmount());
	assertEquals("0.0", parser.getMktPriceMargin());
	assertEquals("25.0", parser.getSecurityDeposit());
```

**Review by sqrrm — APPROVED** [review]
utACK

**Comment by ghubstan** [comment]
Closed with merge of next PR https://github.com/bisq-network/bisq/pull/5284 (without file conflicts).

**Pull request closed without merging**